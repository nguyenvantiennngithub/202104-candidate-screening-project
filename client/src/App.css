body {
    color: #d8d8d8;
    margin: 0;
    box-sizing: border-box;
    font-family: -apple-system, BlinkMacSystemFont, "Roboto", "Segoe UI",
        "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans",
        "Helvetica Neue", sans-serif;
    background-color: #1f1f1f;
}

.container {
    width: 50%;
    margin: 100px auto 100px auto;
    min-height: 500px;
}
