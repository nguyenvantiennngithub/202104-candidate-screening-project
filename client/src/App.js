import "./App.css";
import TodoList from "./component/TodoList/TodoList";
import TodoForm from "./component/TodoForm/TodoForm";
import { useEffect } from "react";
import { useTodo } from "./store/hooks";
import * as todoAction from "./store/todoAction";
import * as todoApi from "./api/todo";
const App = () => {
    const { dispatch } = useTodo();
    useEffect(() => {
        async function getAllTodo() {
            const { data } = await todoApi.getAllTodo();
            dispatch(todoAction.getAllTodo(data.todos));
        }
        getAllTodo();
    }, [dispatch]);
    return (
        <div className="app">
            <div className="container">
                <TodoList></TodoList>
                <TodoForm></TodoForm>
            </div>
        </div>
    );
};

export default App;
