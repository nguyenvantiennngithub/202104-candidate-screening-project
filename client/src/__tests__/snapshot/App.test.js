import App from "../../App";
import Provider from "../../store/TodoProvider";
import renderer from "react-test-renderer";

describe("app", () => {
    it("normal case", () => {
        const tree = renderer.create(
            <Provider>
                <App></App>
            </Provider>
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });
});
