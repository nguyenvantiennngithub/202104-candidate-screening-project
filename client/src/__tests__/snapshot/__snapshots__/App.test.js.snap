// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app normal case 1`] = `
<div
  className="app"
>
  <div
    className="container"
  >
    <div
      className="todoList"
    >
      <div
        className="todoList-header"
      >
        <h2>
          Todo List
        </h2>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1sszw6i-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Clear complated
        </button>
      </div>
    </div>
    <div
      className="todoForm"
    >
      <div
        className="todoForm-content"
      >
        <div>
          <textarea
            className="todoForm-content-input"
            onChange={[Function]}
            placeholder="Enter content"
            value=""
          />
        </div>
        <div
          className="todoForm-addion"
        >
          <div
            className="MuiFormControl-root MuiTextField-root todoForm-date css-nccuzw-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={true}
              htmlFor="datetime-local"
              id="datetime-local-label"
            >
              Due date
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={true}
                id="datetime-local"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="datetime-local"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1in441m"
                >
                  <span>
                    Due date
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <span
            className="MuiCheckbox-root MuiCheckbox-colorSuccess MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSuccess PrivateSwitchBase-root todoForm-checkbox css-10drtbx-MuiButtonBase-root-MuiCheckbox-root"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <input
              checked={false}
              className="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="todoForm-action"
      >
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-ke5b6m-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          id="button-submit"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Add
        </button>
      </div>
    </div>
  </div>
</div>
`;
